Getting error while running on Node 18

Getting error while running `call npm i -g @microsoft/botframework-cli ` on Azure while updating QnA pipeline for chatbot on Azure Platform.

Error:
2023-04-07T11:21:39.3203953Z D:\a\r1\a>call npm i -g @microsoft/botframework-cli 
2023-04-07T11:22:21.6973740Z npm WARN EBADENGINE Unsupported engine {
2023-04-07T11:22:21.6974711Z npm WARN EBADENGINE   package: '@microsoft/bf-orchestrator-cli@4.15.0',
2023-04-07T11:22:21.6975291Z npm WARN EBADENGINE   required: { node: '^10.0.0 || ^12.0.0 || ^13.0.0 || ^14.0.0' },
2023-04-07T11:22:21.6975744Z npm WARN EBADENGINE   current: { node: 'v18.15.0', npm: '9.5.0' }
2023-04-07T11:22:21.6976021Z npm WARN EBADENGINE }
2023-04-07T11:22:21.6986457Z npm WARN EBADENGINE Unsupported engine {
2023-04-07T11:22:21.6987194Z npm WARN EBADENGINE   package: '@microsoft/bf-orchestrator@4.15.0',
2023-04-07T11:22:21.6987789Z npm WARN EBADENGINE   required: { node: '^10.0.0 || ^12.0.0 || ^13.0.0 || ^14.0.0' },
2023-04-07T11:22:21.6988278Z npm WARN EBADENGINE   current: { node: 'v18.15.0', npm: '9.5.0' }
2023-04-07T11:22:21.6988532Z npm WARN EBADENGINE }
2023-04-07T11:22:21.6994344Z npm WARN EBADENGINE Unsupported engine {
2023-04-07T11:22:21.6994943Z npm WARN EBADENGINE   package: '@microsoft/orchestrator-core@4.14.4',
2023-04-07T11:22:21.6995654Z npm WARN EBADENGINE   required: { node: '^10.0.0 || ^12.0.0 || ^13.0.0 || ^14.0.0 || ^16.0.0' },
2023-04-07T11:22:21.6996121Z npm WARN EBADENGINE   current: { node: 'v18.15.0', npm: '9.5.0' }
2023-04-07T11:22:21.6996400Z npm WARN EBADENGINE }
2023-04-07T11:23:00.2376116Z npm WARN deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
2023-04-07T11:23:05.7100772Z npm WARN deprecated @oclif/screen@1.0.4: Deprecated in favor of @oclif/core
2023-04-07T11:23:31.9900573Z npm WARN deprecated cli-ux@4.9.3: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
2023-04-07T11:23:32.5555836Z npm WARN deprecated cli-ux@5.3.3: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
2023-04-07T11:23:33.2113976Z npm WARN deprecated cli-ux@5.6.7: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
2023-04-07T11:24:14.3396691Z npm WARN cleanup Failed to remove some directories [
2023-04-07T11:24:14.3401325Z npm WARN cleanup   [
2023-04-07T11:24:14.3402134Z npm WARN cleanup     'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules',
2023-04-07T11:24:14.3403867Z npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\@azure\ms-rest-js\node_modules\tslib'] {
2023-04-07T11:24:14.3405342Z npm WARN cleanup       errno: -4048,
2023-04-07T11:24:14.3406299Z npm WARN cleanup       code: 'EPERM',
2023-04-07T11:24:14.3406653Z npm WARN cleanup       syscall: 'rmdir',
2023-04-07T11:24:14.3407442Z npm WARN cleanup       path: 'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules\\@azure\\ms-rest-js\\node_modules\\tslib'
2023-04-07T11:24:14.3407895Z npm WARN cleanup     }
2023-04-07T11:24:14.3408109Z npm WARN cleanup   ],
2023-04-07T11:24:14.3408353Z npm WARN cleanup   [
2023-04-07T11:24:14.3408915Z npm WARN cleanup     'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules\\@microsoft',
2023-04-07T11:24:14.3409813Z npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\@microsoft\bf-orchestrator-cli\node_modules\@oclif\config\node_modules\jsonfile'] {
2023-04-07T11:24:14.3410379Z npm WARN cleanup       errno: -4048,
2023-04-07T11:24:14.3410779Z npm WARN cleanup       code: 'EPERM',
2023-04-07T11:24:14.3411171Z npm WARN cleanup       syscall: 'rmdir',
2023-04-07T11:24:14.3411803Z npm WARN cleanup       path: 'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules\\@microsoft\\bf-orchestrator-cli\\node_modules\\@oclif\\config\\node_modules\\jsonfile'
2023-04-07T11:24:14.3412225Z npm WARN cleanup     }
2023-04-07T11:24:14.3412462Z npm WARN cleanup   ],
2023-04-07T11:24:14.3412710Z npm WARN cleanup   [
2023-04-07T11:24:14.3414026Z npm WARN cleanup     'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli',
2023-04-07T11:24:14.3414727Z npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\from2'] {
2023-04-07T11:24:14.3415204Z npm WARN cleanup       errno: -4048,
2023-04-07T11:24:14.3415500Z npm WARN cleanup       code: 'EPERM',
2023-04-07T11:24:14.3415852Z npm WARN cleanup       syscall: 'rmdir',
2023-04-07T11:24:14.3416356Z npm WARN cleanup       path: 'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules\\from2'
2023-04-07T11:24:14.3416689Z npm WARN cleanup     }
2023-04-07T11:24:14.3416886Z npm WARN cleanup   ],
2023-04-07T11:24:14.3417114Z npm WARN cleanup   [
2023-04-07T11:24:14.3417623Z npm WARN cleanup     'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules\\@microsoft\\bf-luis-cli',
2023-04-07T11:24:14.3418070Z npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\@microsoft\bf-luis-cli'] {
2023-04-07T11:24:14.3418751Z npm WARN cleanup       errno: -4048,
2023-04-07T11:24:14.3419055Z npm WARN cleanup       code: 'EPERM',
2023-04-07T11:24:14.3419259Z npm WARN cleanup       syscall: 'rmdir',
2023-04-07T11:24:14.3419608Z npm WARN cleanup       path: 'C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules\\@microsoft\\bf-luis-cli'
2023-04-07T11:24:14.3419837Z npm WARN cleanup     }
2023-04-07T11:24:14.3419989Z npm WARN cleanup   ]
2023-04-07T11:24:14.3420106Z npm WARN cleanup ]
2023-04-07T11:24:14.3443002Z npm ERR! code 1
2023-04-07T11:24:14.3443737Z npm ERR! path C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\@microsoft\orchestrator-core
2023-04-07T11:24:14.3462182Z npm ERR! command failed
2023-04-07T11:24:14.3463086Z npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node-pre-gyp install
2023-04-07T11:24:14.3463926Z npm ERR! response status 404 Not Found on https://bcmodelsprod.azureedge.net/native/orchestrator-core-v4.14.4-node-v108-win32-x64.tar.gz
2023-04-07T11:24:14.3464579Z npm ERR! node-pre-gyp info it worked if it ends with ok
2023-04-07T11:24:14.3465035Z npm ERR! node-pre-gyp info using node-pre-gyp@1.0.10
2023-04-07T11:24:14.3465690Z npm ERR! node-pre-gyp info using node@18.15.0 | win32 | x64
2023-04-07T11:24:14.3466427Z npm ERR! node-pre-gyp info check checked for "C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\@microsoft\orchestrator-core\orchestrator-core.node" (not found)
2023-04-07T11:24:14.3467481Z npm ERR! node-pre-gyp http GET https://bcmodelsprod.azureedge.net/native/orchestrator-core-v4.14.4-node-v108-win32-x64.tar.gz
2023-04-07T11:24:14.3468243Z npm ERR! node-pre-gyp ERR! install response status 404 Not Found on https://bcmodelsprod.azureedge.net/native/orchestrator-core-v4.14.4-node-v108-win32-x64.tar.gz 
2023-04-07T11:24:14.3468835Z npm ERR! node-pre-gyp ERR! install error 
2023-04-07T11:24:14.3469603Z npm ERR! node-pre-gyp ERR! stack Error: response status 404 Not Found on https://bcmodelsprod.azureedge.net/native/orchestrator-core-v4.14.4-node-v108-win32-x64.tar.gz
2023-04-07T11:24:14.3470539Z npm ERR! node-pre-gyp ERR! stack     at C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\@mapbox\node-pre-gyp\lib\install.js:67:15
2023-04-07T11:24:14.3471305Z npm ERR! node-pre-gyp ERR! stack     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
2023-04-07T11:24:14.3471964Z npm ERR! node-pre-gyp ERR! System Windows_NT 10.0.20348
2023-04-07T11:24:14.3472760Z npm ERR! node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\npm\\prefix\\node_modules\\@microsoft\\botframework-cli\\node_modules\\@mapbox\\node-pre-gyp\\bin\\node-pre-gyp" "install"
2023-04-07T11:24:14.3473607Z npm ERR! node-pre-gyp ERR! cwd C:\npm\prefix\node_modules\@microsoft\botframework-cli\node_modules\@microsoft\orchestrator-core
2023-04-07T11:24:14.3474081Z npm ERR! node-pre-gyp ERR! node -v v18.15.0
2023-04-07T11:24:14.3474569Z npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.10
2023-04-07T11:24:14.3475016Z npm ERR! node-pre-gyp ERR! not ok
2023-04-07T11:24:14.3484520Z 
2023-04-07T11:24:14.3487652Z npm ERR! A complete log of this run can be found in:
2023-04-07T11:24:14.3488181Z npm ERR!     C:\npm\cache\_logs\2023-04-07T11_21_41_765Z-debug-0.log
2023-04-07T11:24:14.4495324Z node:internal/modules/cjs/loader:1078
2023-04-07T11:24:14.4495850Z   throw err;
2023-04-07T11:24:14.4496075Z   ^
2023-04-07T11:24:14.4496140Z 
2023-04-07T11:24:14.4496860Z Error: Cannot find module 'C:\npm\prefix\node_modules\@microsoft\botframework-cli\bin\run'
2023-04-07T11:24:14.4497216Z     at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
2023-04-07T11:24:14.4497615Z     at Module._load (node:internal/modules/cjs/loader:920:27)
2023-04-07T11:24:14.4498068Z     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
2023-04-07T11:24:14.4498469Z     at node:internal/main/run_main_module:23:47 {
2023-04-07T11:24:14.4498907Z   code: 'MODULE_NOT_FOUND',
2023-04-07T11:24:14.4499428Z   requireStack: []
2023-04-07T11:24:14.4499589Z }
2023-04-07T11:24:14.4499704Z 
2023-04-07T11:24:14.4499873Z Node.js v18.15.0
2023-04-07T11:24:14.5261469Z node:internal/modules/cjs/loader:1078
2023-04-07T11:24:14.5261897Z   throw err;
2023-04-07T11:24:14.5262142Z   ^
2023-04-07T11:24:14.5262209Z 
2023-04-07T11:24:14.5262798Z Error: Cannot find module 'C:\npm\prefix\node_modules\@microsoft\botframework-cli\bin\run'
2023-04-07T11:24:14.5263202Z     at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
2023-04-07T11:24:14.5263596Z     at Module._load (node:internal/modules/cjs/loader:920:27)
2023-04-07T11:24:14.5263940Z     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
2023-04-07T11:24:14.5264154Z     at node:internal/main/run_main_module:23:47 {
2023-04-07T11:24:14.5264406Z   code: 'MODULE_NOT_FOUND',
2023-04-07T11:24:14.5264579Z   requireStack: []
2023-04-07T11:24:14.5264683Z }
2023-04-07T11:24:14.5264783Z 
2023-04-07T11:24:14.5264883Z Node.js v18.15.0
2023-04-07T11:24:14.5413991Z ##[error]Script failed with exit code: 1

--- a/packages/orchestratorlib/package.json
+++ b/packages/orchestratorlib/package.json
@@ -37,10 +37,10 @@
   "dependencies": {
     "@microsoft/bf-lu": "1.0.0",
     "@microsoft/bf-dispatcher": "1.0.0",
-    "axios":"~0.21.4",
+    "axios": "~0.21.4",
     "https-proxy-agent": "^5.0.0",
     "tslib": "^2.0.3",
-    "@microsoft/orchestrator-core": "~4.14.0",
+    "@microsoft/orchestrator-core": "https://bcmodelsprod.azureedge.net/native/orchestrator-core-v4.14.4-node-v93-win32-x64.tar.gz",
     "@types/fs-extra": "~8.1.0",
     "fs-extra": "~9.0.0",
     "read-text-file": "~1.1.0",

--- a/rush.json
+++ b/rush.json
@@ -1,6 +1,6 @@
 {
   "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/rush.schema.json",
-  "rushVersion": "5.19.1",
+  "rushVersion": "5.33.0",
   "pnpmVersion": "4.14.4",
   "pnpmOptions": {},
   "nodeSupportedVersionRange": ">=14.0.0",

